cmd/mkroot/commands.go

```diff
@@ -174,125 +148,141 @@ func useLocalInit(repo string) (initPath string, runConfig []byte, err error) {
-		}
+	initDir, rmInit, err := mktemp()
+	if err != nil {
+		return err
 	}
 
-	if m.makeInit {
-		var (
-			initPath string
-			runJson  []byte
-			err      error
-		)
-
-		if !Exists(m.initFrom) {
-			if initPath, runJson, err = installInitFromGithub(); err != nil {
-				return err
-			}
-		} else {
-			abs, err := filepath.Abs(m.initFrom)
-			if err != nil {
-				return fmt.Errorf("failed get absolute path of local repo: %v", err)
-			}
-
-			if initPath, runJson, err = useLocalInit(abs); err != nil {
-				return err
-			}
+	defer func() {
+		if err := rmInit(); err != nil {
+			log.Fatalf("failed to remove temp dir: '%s': %v", initDir, err)
 		}
+	}()
 
-		initDir, rmDir, err := mktemp()
-		if err != nil {
-			return err
-		}
+	configDir := filepath.Join(initDir, "thi")
 
-		defer func() {
-			if err := rmDir(); err != nil {
-				log.Fatalf("failed to remove temp dir: '%s': %v", initDir, err)
-			}
-		}()
+	if err := os.MkdirAll(configDir, 0755); err != nil {
+		return fmt.Errorf("failed to create config dir in init drive: %v", err)
+	}
 
-		configDir := filepath.Join(initDir, "thi")
+	if err := copyFile(initPath, filepath.Join(configDir, "init")); err != nil {
+		return fmt.Errorf("failed to copy init: %v", err)
+	}
 
-		if err := os.MkdirAll(configDir, 0755); err != nil {
-			return fmt.Errorf("failed to create config dir in init drive: %v", err)
-		}
+	if err := os.WriteFile(filepath.Join(configDir, "run.json"), runJson, 0755); err != nil {
+		return fmt.Errorf("failed to create run.json file: %v", err)
+	}
+	m.from = initDir
 
-		if err := copyFile(initPath, filepath.Join(configDir, "init")); err != nil {
-			return fmt.Errorf("failed to copy init: %v", err)
-		}
+	temp, rmTemp, err := mktemp()
+	if err != nil {
+		return fmt.Errorf("failed to make temporary directory: %v", err)
+	}
 
-		if err := os.WriteFile(filepath.Join(configDir, "run.json"), runJson, 0755); err != nil {
-			return fmt.Errorf("failed to create run.json file: %v", err)
+	defer func() {
+		if err := rmTemp(); err != nil {
+			log.Fatalf("failed to remove temp dir: '%s': %v", temp, err)
 		}
-		m.from = initDir
+	}()
+
+	mount := exec.Command(mountPath, "-o", m.mountArgs, m.name, temp)
+	if err := mount.Run(); err != nil {
+		return fmt.Errorf("failed to mount '%s': %v", m.name, err)
 	}
 
-	if m.makeInit || Exists(m.from) {
-		temp, rmDir, err := mktemp()
-		if err != nil {
-			return fmt.Errorf("failed to make temporary directory: %v", err)
-		}
+	if err := copy.DirCopy(m.from, temp, copy.Content, true); err != nil {
+		return fmt.Errorf("failed to copy from '%s' to '%s': %v", m.from, temp, err)
+	}
+
+	umount := exec.Command(umountPath, temp)
+	if err := umount.Run(); err != nil {
+		return fmt.Errorf("failed to unmount '%s': %v", temp, err)
+	}
+	return nil
+}
 
-		defer func() {
-			if err := rmDir(); err != nil {
-				log.Fatalf("failed to remove temp dir: '%s': %v", temp, err)
-			}
-		}()
+func (m *mkfs) createDriveFromImage() error {
```

Comment:
this function contains code for creating the loop drive with the specified image.